jplag/src/main/java/de/jplag/CommandLineArgument.java

```diff
@@ -21,40 +21,43 @@
  * @author Timur Saglam
  */
 public enum CommandLineArgument {
-    ROOT_DIRECTORY("rootDir", String.class),
+    ROOT_DIRECTORIES("rootDir", Optional.of("+"), String.class),
     LANGUAGE("-l", String.class, LanguageOption.getDefault().getDisplayName(), LanguageOption.getAllDisplayNames()),
-    BASE_CODE("-bc", String.class),
+    BASE_CODE("-bc", Optional.empty(), String.class),
     VERBOSITY("-v", String.class, "quiet", List.of("quiet", "long")), // TODO SH: Replace verbosity when integrating a real logging library
-    DEBUG("-d", Boolean.class),
-    SUBDIRECTORY("-S", String.class),
-    SUFFIXES("-p", String.class),
-    EXCLUDE_FILE("-x", String.class),
-    MIN_TOKEN_MATCH("-t", Integer.class),
+    DEBUG("-d", Optional.empty(), Boolean.class),
+    SUBDIRECTORY("-S", Optional.empty(), String.class),
+    SUFFIXES("-p", Optional.empty(), String.class),
+    EXCLUDE_FILE("-x", Optional.empty(), String.class),
+    MIN_TOKEN_MATCH("-t", Optional.empty(), Integer.class),
     SIMILARITY_THRESHOLD("-m", Float.class, DEFAULT_SIMILARITY_THRESHOLD),
     SHOWN_COMPARISONS("-n", Integer.class, DEFAULT_SHOWN_COMPARISONS),
     RESULT_FOLDER("-r", String.class, "result"),
     COMPARISON_MODE("-c", String.class, DEFAULT_COMPARISON_MODE.getName(), ComparisonMode.allNames());
 
     private final String flag;
+    private final Optional<String> nArgs;
     private final String description;
     private final Optional<Object> defaultValue;
     private final Optional<Collection<String>> choices;
     private final Class<?> type;
 
-    private CommandLineArgument(String flag, Class<?> type) {
-        this(flag, type, Optional.empty(), Optional.empty());
+    private CommandLineArgument(String flag, Optional<String> nArgs, Class<?> type) {
```

Comment:
It is here, next patch uses both `*` and `+`. Use the string as defined by the argparse4j package or should I add an enum definition?

Reply:
I would prefer the enum. I actually think argparse4j should have provided that enum for us instead of taking a parameter of type string. But if we need both * and + soon, then let us make an enum.

Reply:
`argparse4j` is sticking very close to Python where you neither have enumerations nor constants. It looks like a Python programmer writing Java to me.

The library has more weak spots, retrieving data from `NameSpace` is somewhat ieky, as returned data changes type depending on specified settings of a commandline argument.